### DISCUSSION TITLE
Segmentation Fault with different starts or ends

### DISCUSSION THREAD

**Discussion by FreddyHunter** [google/or-tools#2573] [Routing (and legacy CP) questions]
Hi Community Guys,

I have been using this fantastic tool for a while, but I am facing some difficulties when trying to model vehicles to start or to finish at different locations. I already checked the examples and others opened issues related, but I am still getting Segmentation Fault as error message. And it is not clear for me where is the issue. For example, if I set starts with zeros and ends with zeros it run, if I set starts with node numbers and ends with the same node numbers, it also run. But, if I set starts with zeros and ends with node numbers, or the opposite, it does not run and return Segmentation Fault.

Does someone have any tip or recommendation to check?

Note: I am using the latest python version.

Thanks in advance for your attention.

**Comment by Mizux** [comment]
my 2 cents
`manager.IndexToNode()` is defined for all index while `manager.NodeToIndex()` is not defined for start and end node -> did you use `manager.NodeToIndex()` with a start or end node as parameter ?

**Comment by FreddyHunter** [comment]
Hi Mizux,

Thanks for your quick reply, but I am using manager.IndexToNode (). Is just
replace this one for Nodetoindex? How do I pass the parameters?

Thanks.

**Reply by Mizux** [reply]
no you don't have to replace.
I just wanted to point out that often segfaults come from an usage of `NodeToIndex()` with a node being a start or an end.

Also do you have a trace or something to show ? do you know which line failed ? is it during the model creation ? during the solve ? when printing the solution ?

**Comment by FreddyHunter** [comment]
Hi Mizux,

The solver even start when I have starts or ends different than zeros, just
give me the error. Is there a way to print a trace or log to help us
understand the issue? I just know how to print the log search.

Basically, I modeled it in the same way of different starts and ends
example, which I used to test just different starts, or just different ends
and etc. The difference is my model considers capacity constraints, time
windows, pickups and deliveries, different cargo types, resources
constraints, fixed and variable costs per vehicles, max stops per route,
max distance per vehicle, which is running pretty well. Just different
starts and ends I was not able to model together.

Thanks for your help.

**Comment by FreddyHunter** [comment]
Hi Mizux,
any other tips or suggestions because I am getting segmentation fault?
Thanks.
